tracing-subscriber/src/registry/fmt.rs

```diff
@@ -0,0 +1,349 @@
+}
+
+impl<S, N, E, W> fmt::Debug for FmtLayer<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    E: FormatEvent<S, N> + 'static,
+    W: MakeWriter + 'static,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("FmtLayer").finish()
+    }
+}
+
+/// A builder for `FmtLayer` that logs formatted representations of `tracing` events.
+pub struct FmtLayerBuilder<
+    S = Registry,
+    N = format::DefaultFields,
+    E = format::Format<format::Full>,
+    W = fn() -> io::Stdout,
+> {
+    fmt_fields: N,
+    fmt_event: E,
+    make_writer: W,
+    is_interested: Box<dyn Fn(&Event<'_>) -> bool + Send + Sync + 'static>,
+    _inner: PhantomData<S>,
+}
+
+impl<S, N, E, W> fmt::Debug for FmtLayerBuilder<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    E: FormatEvent<S, N> + 'static,
+    W: MakeWriter + 'static,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("FmtLayerBuilder").finish()
+    }
+}
+
+impl FmtLayer {
+    /// Creates a [FmtLayerBuilder].
+    pub fn builder() -> FmtLayerBuilder {
+        FmtLayerBuilder::default()
+    }
+}
+
+impl<S, N, E, W> FmtLayerBuilder<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    E: FormatEvent<S, N> + 'static,
+    W: MakeWriter + 'static,
+{
+    /// Sets a filter for events. This filter applies to this, and
+    /// subsequent, layers.
+    pub fn with_interest<F>(self, f: F) -> Self
+    where
+        F: Fn(&Event<'_>) -> bool + Send + Sync + 'static,
+    {
+        FmtLayerBuilder {
+            fmt_fields: self.fmt_fields,
+            fmt_event: self.fmt_event,
+            make_writer: self.make_writer,
+            is_interested: Box::new(f),
+            _inner: self._inner,
+        }
+    }
+}
+
+// This, like the MakeWriter block, needs to be a seperate impl block because we're
+// overriding the `E` type parameter with `E2`.
+impl<S, N, E, W> FmtLayerBuilder<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    E: FormatEvent<S, N> + 'static,
+    W: MakeWriter + 'static,
+{
+    /// Sets a [FormatEvent<S, N>].
+    pub fn with_event_formatter<E2>(self, e: E2) -> FmtLayerBuilder<S, N, E2, W> {
+        FmtLayerBuilder {
+            fmt_fields: self.fmt_fields,
+            fmt_event: e,
+            make_writer: self.make_writer,
+            is_interested: self.is_interested,
+            _inner: self._inner,
+        }
+    }
+}
+
+// this needs to be a seperate impl block because we're re-assigning the the W2 (make_writer)
+// type paramater from the default.
+impl<S, N, E, W> FmtLayerBuilder<S, N, E, W> {
+    /// Sets a [MakeWriter] for spans and events.
```

Comment:
link needs a reference, and backticks.

Reply:
also, i believe that the `FmtSubscriber`'s builder had a doc comment for this with an example, can we put that in here?